libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1230,30 +1231,17 @@ bool IRGeneratorForStatements::visit(Literal const& _literal)
 	return false;
 }
 
-void IRGeneratorForStatements::appendExternalFunctionCall(
-	FunctionCall const& _functionCall,
-	vector<ASTPointer<Expression const>> const& _arguments
-)
+IRGeneratorForStatements::ReturnInfo IRGeneratorForStatements::collectReturnInfo(FunctionCall const& _functionCall)
```

Comment:
Is it possible to use the same logic also in the old code generator?

I mean we don't have to store all the derived booleans, but maybe we can create a small class with functions that compute them? My main concern is that we duplicate logic and then change it at one place but forget to change it at the other.